Jesse Moore wip

jesse-moore opened the pull request.

ericwburden reviewed (commented):
This certainly passes the given tests, and I like the approach of iterating by character and keeping a 'word buffer'. 

I have one concern with the way you're counting characters, though. What do you get for `InterveningVerbiage.countWordsBetween("a", "snack", "I would do anything for a delicious snack");` For a future refactor, I'd also suggest that this function has enough complexity that I'd start thinking about how to abstract pieces of it out to other classes/functions.

That said, this is some great work, and really demonstrates some clever problem-solving!

ericwburden commented:
I have an answer!

![image](https://user-images.githubusercontent.com/22438085/114038944-eb236f00-9847-11eb-9a91-7a3f008f0bd8.png)

I "think" what is happening has to do with the way the buffer is being created, where "anything" is throwing off the search for "a".  Here's the test case: 

```java
/**
     * Extra test case
     */
    @Test
    @DisplayName("Words between 'a' and 'snack'")
    public void testCaseExtra() {
        String sentence = "I would do anything for a delicious snack";
        String msg = "One word between 'a' and 'snack'";
        try {
            int result = testClass.countWordsBetween("a", "snack", sentence);
            assertEquals(1, result, msg);
        } catch (Exception e) {
            fail("Should not throw an exception for valid arguments.");
        }
    }
```

jesse-moore commented:
Thanks for the feedback, the issue was I wasn't checking if the partial matching word was longer than the start word.

line 60 should be this
`if (!isCounting && keywordIndex<start.length() && charEqualsIgnoreCase(words.charAt(i),start.charAt(keywordIndex))) keywordIndex++;`

instead of this
`if (!isCounting && charEqualsIgnoreCase(words.charAt(i),start.charAt(keywordIndex))) keywordIndex++;`

I refactored it and fixed the issue.
